### PR TITLE
sync public channels from rhr widget

### PR DESCRIPTION
**Pull request by camhashemi**
@cvializ @andrewconner still needs some QA before releasing

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/home/featureUpsell.tpl.html [hunk 1]
```diff
@@ -9,7 +9,7 @@
       <span class="kf-fu-body-line">Search within Slack using <code>/kifi</code>.</span>
       <span class="kf-fu-body-line">Get your team on Kifi easily.</span>
     </div>
-    <a class="kf-fu-slack-start-btn" target="_self" href="/integrations/slack/start" ng-click="clickedConnectSlack()">Connect Slack</a>
-    <a class="kf-fu-slack-learn-link kf-link" href="https://www.kifi.com/integrations/slack?loggedIn" target="_self">Learn more</a>
+    <a class="kf-fu-slack-start-btn" ng-click="clickedConnectSlack()">Connect Slack</a>
+    <a class="kf-fu-slack-learn-link kf-link" href="https://www.kifi.com/integrations/slack?loggedIn">Learn more</a>
```
Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/home/featureUpsell.tpl.html:13]: I think you might still want `target="_self"` for the Learn more

File: kifi-backend/modules/shoebox/angular/src/home/featureUpsell.js [hunk 2]
```diff
@@ -14,11 +14,14 @@ angular.module('kifi')
       templateUrl: 'home/featureUpsell.tpl.html',
       link: function (scope) {
         scope.me = profileService.me;
-        var hasFeatureUpsellExp = (profileService.me.experiments || []).indexOf('slack_upsell_widget') !== -1;
+        var hasFeatureUpsellExp = (scope.me.experiments || []).indexOf('slack_upsell_widget') !== -1;
+        var orgToSync = scope.me.orgs.filter(function (org) {
+          return !org.slackTeam;
+        })[0];
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/angular/src/home/featureUpsell.js:20]: This won't include orgs that have connected a Slack team but have not synced their channels yet, intended?

File: kifi-backend/modules/shoebox/angular/src/home/featureUpsell.tpl.html (hunk 1 above)
Comment (leogrim) [inline kifi-backend/modules/shoebox/angular/src/home/featureUpsell.tpl.html:13]: I've never seen this `loggedIn` query param, what is it?

File: kifi-backend/modules/shoebox/angular/src/home/featureUpsell.js (hunk 2 above)
Comment (camhashemi) [inline kifi-backend/modules/shoebox/angular/src/home/featureUpsell.js:20]: I think at the time this was created you only had a slack team if you integrated all your channels, and the spec has been lost over time. We can do whatever we want. I think we should include the orgs you mentioned if the goal is to sync all channels. If the goal is to just connect your team, we can keep the same orgs and change the action.

File: kifi-backend/modules/shoebox/angular/src/home/featureUpsell.tpl.html (hunk 1 above)
Comment (cvializ) [inline kifi-backend/modules/shoebox/angular/src/home/featureUpsell.tpl.html:13]: it's passed to the marketing page to decide whether or not to render a login prompt

Comment (camhashemi) [inline kifi-backend/modules/shoebox/angular/src/home/featureUpsell.tpl.html:13]: the login header (after scrolling) still shows up with it, so I'll just take it out.

**Comment by camhashemi** [comment]
updated. specs: show if the user has not touched the widget before and has an org without a slack integration or without a public sync. if the user clicks "sync" or "hide" on the widget, we don't show it again. The specs for this may be outdated / piecemeal so if @leogrim or @JRuff42 could approve it or suggest polish (e.g. show whenever an org can be sync'd, don't hide it after the first time), that'd be great.

**Comment by camhashemi** [comment]
screen shot 2016-03-04 at 2 40 42 pm

screen shot 2016-03-04 at 2 40 55 pm

File: kifi-backend/modules/shoebox/angular/src/home/featureUpsell.tpl.html (hunk 1 above)
Comment (camhashemi) [inline kifi-backend/modules/shoebox/angular/src/home/featureUpsell.tpl.html:13]: actually, probably better to fix what's broken, so I'm leaving it in and will make a small asana task for it.

**Pull request merged**